src/VASL/build/module/ASLMap.java

```diff
@@ -175,6 +176,14 @@ public JPopupMenu getPopupMenu()
     @Override
     public synchronized void setBoards(Collection<Board> c) {
 
+        Iterator it = c.iterator();
+        while (it.hasNext()) {

```

Comment:
I would never use an `Iterator` for iterating over a whole collection; it's simpler to use a for-each loop:
```
for (VASLBoard testboardexists: c) {
  // do stuff
}
```
It's two lines shorter and involves no casting.

Reply:
Otherwise, this seems like a sensible solution to your problem.

Reply:
I have replaced the iterator with a for-each loop; no problem with that. 

However, for (VASLBoard testboardexists: c) { produces a type error as c is a collection of Board. So, I had to keep the cast. 

I have pushed up a new commit (see 5f9d734)